src/config.c

```diff
@@ -261,8 +312,7 @@ restraint_config_set (gchar *config_file, const gchar *section,
                                        g_value_get_string (&value));
                 break;
             default:
-                g_warning ("invalid GType\n");
-                goto error;
+                g_error ("%s (): invalid GType", __func__);
```

Comment:
I don't think we should reintroduce g_error into restraint code.

Reply:
Could you elaborate why?

Reply:
Because it will result in a core dump.

Reply:
As I wrote in the commit message,
```
fix: Make invalid type fatal in restraint_config_set

Passing an invalid type to restraint_config_set is a programming error.

Use g_error as assertion.

Note that g_error calls will terminate the application.
```

Reply:
I think the original warning should suffice and the program should move on.  We removed g_error in similar situation and it doesn't make sense to add it back in.

Reply:
So, programming errors means, for example, that a developer wrote a `restraint_config_set ()` call passing an invalid type, or, handling for a given type is removed from `restraint_config_set ()` and calls in the program are not updated accordingly. The original warning only suffices to hide these issues. A call like this shouldn't pass testing, but we still lack coverage.

Reply:
Another concern for adding it back in is someone may see it and decide to use it else where not expecting it to core dump.  I've frankly never seen an error reporting function to do that unless it was explicitly named something like g_abort.